internal/pkg/agent/application/monitoring/process.go

```diff
@@ -5,25 +5,45 @@
 package monitoring
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"net/http"
+	"strings"
+	"syscall"
+	"time"
 
 	"github.com/gorilla/mux"
 
 	"github.com/elastic/elastic-agent/internal/pkg/agent/application/coordinator"
 	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
 )
 
-const processIDKey = "processID"
+const (
+	processIDKey      = "processID"
+	timeout           = 10 * time.Second
+	fleetServerPrefix = "fleet-server"
+)
+
+var redirectPathAllowlist = map[string]struct{}{
+	"stats": {},
+	"state": {},
+}
+
+var redirectableProcesses = []string{
+	"apm-server",
+	fleetServerPrefix,
+}
 
-func processHandler(coord *coordinator.Coordinator, statsHandler func(http.ResponseWriter, *http.Request) error) func(http.ResponseWriter, *http.Request) error {
+func processHandler(coord *coordinator.Coordinator, statsHandler func(http.ResponseWriter, *http.Request) error, operatingSystem string) func(http.ResponseWriter, *http.Request) error {
 	return func(w http.ResponseWriter, r *http.Request) error {
 		w.Header().Set("Content-Type", "application/json; charset=utf-8")
 
 		vars := mux.Vars(r)
-		id, found := vars[processIDKey]
 
+		id, found := vars[processIDKey]
 		if !found {
 			return errorfWithStatus(http.StatusNotFound, "productID not found")
 		}
```

Comment:
[Blockerish]
Why the code uses _process_ and the returned error, _product_? Is it intentional? It looks like a mistake or typo

Reply:
that's the fine question :D